src/geocat/comp/climatology.py

```diff
@@ -322,3 +352,140 @@ def month_to_season(
+        'season': (None, 'QS-DEC', 'MS'),
+        'year': (None, 'YS', '6MS')
+    }
+
+    try:
+        freq_dict[freq]
+    except KeyError:
+        raise KeyError(
+            f"contributed: calendar_average: bad period: PERIOD = {freq}. Valid periods include: {list(freq_dict.keys())}"
+        )
+
+    # If freq is 'season', key is set to monthly in order to calculate monthly
+    # averages which are then used to calculate seasonal averages
+    key = 'month' if freq in {'season', 'year'} else freq
+
+    (format, frequency, offset) = freq_dict[key]
+
+    time_dim = _get_time_coordinate_info(dset, time_dim)
+
+    # Retrieve calendar name
+    if calendar is None:
+        calendar = xr.coding.times.infer_calendar_name(dset[time_dim])
+
+    if freq == 'year' and climatology:
+        climatology = False
+        warnings.warn(
+            'Cannot compute yearly climatology data since climatologies are averaged over years. To remove warning, set `climatology` to False'
+        )
+    # Average data across years
+    if climatology:
+        if freq == 'season':
+            # Calculate monthly average before calculating seasonal climatologies
+            dset = dset.resample({time_dim: frequency}).mean().dropna(time_dim)
+
+            # Compute the weights for the months in each season so that the
+            # seasonal averages account for months being of different lengths
+            month_length = dset[time_dim].dt.days_in_month.groupby(
+                f"{time_dim}.season")
+            weights = month_length / month_length.sum()
+            dset = (dset * weights).groupby(f"{time_dim}.season")
+            dset = dset.sum(dim=time_dim)
+        else:
+            # Retrieve floor of median year
+            median_yr = np.median(dset[time_dim].dt.year.values)
+
+            # Group data by format then calculate average of groups
+            dset = dset.groupby(
+                dset[time_dim].dt.strftime(format)).mean().rename(
+                    {'strftime': time_dim})
+
+            # Create array of datetimes to set as time coordinate of returned data
+            # Offsets are used to ensure the time coordinate of the returned climatology is centered on the period
+            start_time = dset[time_dim].values[0]
+            end_time = dset[time_dim].values[-1]
+
+            dset = _calculate_center_of_time_bounds(
+                dset,
+                time_dim,
+                frequency,
+                calendar,
+                start=f'{median_yr:.0f}-{start_time}',
+                end=f'{median_yr:.0f}-{end_time}')
+
+    # Average data for each period considering the year of the period
+    else:
+        # Resample data using given frequency which preserves the year of the data
+        dset = dset.resample({time_dim: frequency}).mean().dropna(time_dim)
+        if freq == 'season' or freq == 'year':
+            key = freq
+            (format, frequency, offset) = freq_dict[key]
```

Comment:
`offset` isn't used after this line. 

```suggestion
            format, frequency, _ = freq_dict[key]
```

Reply:
`offset` is no longer used, and I had forgotten to remove it from this line and from the dictionary. It has been fixed in the latest commit along with adding your formatting changes.